src/pint/fitter.py

```diff
@@ -1527,5 +1532,9 @@ def fit_toas(self, maxiter=1, threshold=0, full_cov=False):
                 self.resids.noise_resids = noise_resids
 
         self.update_model(chi2)
+        self.resids.chi2 = chi2
+
+        # A cheat for now:
+        self.resids.toa.chi2 = self.resids.chi2 - self.resids.dm.chi2
```

Comment:
You could just use `chi2-self.resids.dm.chi2` here and avoid needing a setter in CombinedResiduals.

Reply:
You're right, given the other comment.